src/coap_session.c

```diff
@@ -2214,6 +2214,15 @@ coap_session_get_state(const coap_session_t *session) {
   return 0;
 }
 
+coap_endpoint_t *
+coap_session_get_endpoint(const coap_session_t *session) {
+#if COAP_SERVER_SUPPORT
+  if (session)
+    return session->endpoint;
+#endif
```

Comment:
This should read (and removes compilation warning errors about unused parameters)
````
#if COAP_SERVER_SUPPORT
  if (session)
    return session->endpoint;
#else /* ! COAP_SERVER_SUPPORT */
  (void)session;
#endif /* ! COAP_SERVER_SUPPORT */
````